drivers/serial/uart_ns16550.c

```diff
@@ -622,6 +622,11 @@ static int uart_ns16550_configure(const struct device *dev,
 			goto out;
 		}
 
+		if (clock_control_on(dev_cfg->clock_dev, dev_cfg->clock_subsys)) {
```

Comment:
The existing code in this driver is consistently using `if (foo() != 0)` to check for errors (just see a few lines further below). I think it'd be good to follow that practice. Not treating integers as boolean is also required to fulfill rule 85 of the [Coding Guidelines](https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html): "The controlling expression of an if statement and the controlling expression of an iteration-statement shall have essentially Boolean type"

(note: this is only enforced for the subset of Zephyr that's under the safety certification scope, however it's good to follow those for all Zephyr code whenever possible)